src/main/java/seedu/expense/model/ExpenseBook.java

```diff
@@ -289,8 +290,31 @@ public Amount tallyBalance() {
         return tallyBudgets().subtract(tallyExpenses());
     }
 
+    @Override
+    public String getBudgetBarLabel() {
+        ArrayList<Tag> checkedTagList = new ArrayList<>();
+        tags.getTags().stream().map(tag -> Expense.getGenericExpenseWithTag(tag))
+                .filter(expense -> expenses.getFilteredList().getPredicate().test(expense))
+                .map(expense -> expense.getTag()).forEach(checkedTagList::add);
+        return getBudgetLabelFromList(checkedTagList);
+    }
+
```

Comment:
I think it is quite an overkill to create methods in each of the Expense fields just for this. Have you tried using a Set to collect the Tags from the filtered expenses?

e.g.
```
expenses.getFilteredList().stream()
        .map(expense -> expense.getTag())
        .collect(Collectors.toSet()) // remove duplicates
        .stream()
        .forEach(checkedTagList::add);
```